cmd/kubeadm/app/phases/certs/renewal/readwriter.go

```diff
@@ -183,6 +199,12 @@ func (rw *kubeConfigReadWriter) Write(newCert *x509.Certificate, newKey crypto.S
 		return errors.Wrapf(err, "failed to marshal private key to PEM")
 	}
 
+	// Update the embedded CA in the kubeconfig file.
+	// This assumes that the user has kept the current context to the desired one.
+	clusterName := rw.kubeConfig.Contexts[rw.kubeConfig.CurrentContext].Cluster
+	cluster := rw.kubeConfig.Clusters[clusterName]
+	cluster.CertificateAuthorityData = pkiutil.EncodeCertPEM(rw.caCert)
```

Comment:
This looks a bit scary. Perhaps log the overwrite somewhere if the old and new bytes differ?
Other than that, it looks good to me.

/lgtm
/hold
for @fabriziopandini to TAL.

Reply:
well we are using the same CA file for the renewal of certificates.
using a comparison first is fine, but it will not gain security benefits, only a performance benefit.

Reply:
Yep, I guess I am over paranoid here. It's fine like it is.